lib/engine/game/g_18_esp/game.rb

```diff
@@ -1143,6 +1143,20 @@ def start_merge(corporation, minor, keep_token)
 
           # close corp
           close_corporation(minor)
+
+          # delete mz slot if mz
+          fix_cheater_slot_madrid(minor.city) if minor.id == 'MZ'
+        end
+
+        def fix_cheater_slot_madrid(slot)
+          cities = hex_by_id('F24')&.tile&.cities
+          return unless cities
+
+          city = cities[slot]
+          return unless city
+
+          tokens = city.tokens
+          tokens.compact! if tokens.size > city.normal_slots
```

Comment:
Would it be possible for MZ to close when the Madrid cities have merged and there are slots with reservations? This could cause the ordering of tokens and reservations to go wrong.